### PR TITLE
Vibhansa/stream crash

### PR DESCRIPTION
**Pull request by vibhansa-msft**
#674 : Set content type correctly for files > 64MB in size
#675 : SegFault in streaming when open and close come in parallel for same file.

### CODE REVIEW

File: blobfuse/include/BlobStreamer.h [hunk 1]
```diff
@@ -87,24 +87,18 @@ class StreamObject {
         }
 
         int IncRefCount() {
```
Comment (gapra-msft) [inline blobfuse/include/BlobStreamer.h:89]: Should we add a comment here to mention that users of this method must hold the lock before calling this? For this method, Dec and GetRefCount

File: blobfuse/include/BlobStreamer.h [hunk 2]
```diff
@@ -87,24 +87,18 @@ class StreamObject {
...
             return ref_count;
         }
 
         int DecRefCount() {
-            std::lock_guard<std::mutex> lock(m_mutex);
+            //std::lock_guard<std::mutex> lock(m_mutex);
             ref_count--;
-
-            if (ref_count == 0) {
-                Cleanup();
```
Comment (gapra-msft) [inline blobfuse/include/BlobStreamer.h:100]: Was this call to Cleanup unnecessary cause we have code that handles it in BlobStreamer.cpp line 214?

**Review by gapra-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

File: blobfuse/include/BlobStreamer.h (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/include/BlobStreamer.h:100]: It was redundant, in closeFile flow we were decrementing the count and also it was checking if its 0 then cleanup. Hence removed a implicit cleanup.

File: blobfuse/include/BlobStreamer.h (hunk 1 above)
Comment (vibhansa-msft) [reply blobfuse/include/BlobStreamer.h:89]: Will add comments

**Pull request merged**